src/Cognito/Provider.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace SocialiteProviders\Cognito;
+
+use Illuminate\Support\Arr;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'COGNITO';
+
+    /**
+     * Scopes defintions.
+     */
+    public const SCOPE_OPENID = 'openid';
+    public const SCOPE_EMAIL = 'email';
+    public const SCOPE_PHONE = 'phone';
+    public const SCOPE_PROFILE = 'profile';
+    public const SCOPE_ADMIN = 'aws.cognito.signin.user.admin';
+
+    /**
+     * {@inheritdoc}
+     * Adjust the available read / write attributes in cognito client app.
+     */
+    protected $scopes = [
+        self::SCOPE_OPENID,
+        self::SCOPE_PROFILE,
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function additionalConfigKeys()
+    {
+        return [
+            'host',
+            'authorize_uri',
+            'token_uri',
+            'userinfo_uri',
+            'userinfo_key',
+            'user_id',
+            'user_nickname',
+            'user_name',
+            'user_email',
+            'user_avatar',
+        ];
+    }
+
+    /**
+     * Get the authentication URL for the provider.
+     *
+     * @param string $state
+     *
+     * @return string
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase($this->getCognitoUrl('authorize_uri'), $state);
+    }
+
+    /**
+     * Get the token URL for the provider.
+     *
+     * @return string
+     */
+    protected function getTokenUrl()
+    {
+        return $this->getCognitoUrl('token_uri');
+    }
+
+    /**
+     * Get the raw user for the given access token.
+     *
+     * @param string $token
+     *
+     * @return array
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get($this->getCognitoUrl('userinfo_uri'), [
+            'headers' => [
```

Comment:
Can you use `RequestOption::HEADERS` here instead?

Reply:
Done.